### PR TITLE
DTSSTCI-696

### PR DESCRIPTION
**Pull request by gmmagruder**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSSTCI-696

### Change description ###
Add jenkins pipeline

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: charts/sptribs-wa-post-deployment-ft-tests/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+appVersion: "1.0"
+description: A Helm chart for sptribs-wa-post-deployment-ft-tests App
+name: sptribs-wa-post-deployment-ft-tests
+home: https://github.com/hmcts/sptribs-wa-post-deployment-ft-tests
+version: 0.0.4
+maintainers:
+  - name: HMCTS sptribs team
+dependencies:
+  - name: java
+    version: 4.0.11
```
Comment (tomxelliott) [inline charts/sptribs-wa-post-deployment-ft-tests/Chart.yaml]: This is a very old chart - can you upgrade to 5.2.1

File: src/functionalTest/java/uk/gov/hmcts/reform/wapostdeploymentfttests/ScenarioRunnerTest.java [hunk 2]
```diff
@@ -118,6 +119,7 @@ public void setUp() {
     }
 
     @Test
+    @Ignore
     public void scenarios_should_behave_as_specified() throws Exception {
```
Comment (tomxelliott) [inline src/functionalTest/java/uk/gov/hmcts/reform/wapostdeploymentfttests/ScenarioRunnerTest.java]: Minor one but can we tweak this test name so it adheres to correct camelCase standards

File: src/functionalTest/resources/application-functional.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ wa-post-deployment-test:
 azure:
   servicebus:
     connection-string: ${AZURE_SERVICE_BUS_CONNECTION_STRING}
-    topic: ${AZURE_SERVICE_BUS_TOPIC_NAME:ccd-case-events-aat}
+    topic: st-ccd-case-event
```
Comment (tomxelliott) [inline src/functionalTest/resources/application-functional.yaml:18]: is this the correct topic name? where did you get the value from?

Comment (gmmagruder) [reply src/functionalTest/resources/application-functional.yaml:18]: It was what was set before in the build.gradle file. I tried to use what had been used in the PR below but this was the only value that got the functional tests to run.

https://github.com/hmcts/sptribs-case-api/pull/1773

File: charts/sptribs-wa-post-deployment-ft-tests/Chart.yaml [hunk 4]
```diff
@@ -8,5 +8,5 @@ maintainers:
   - name: HMCTS sptribs team
 dependencies:
   - name: java
-    version: 4.0.11
+    version: 5.2.1
     repository: '@hmctspublic'
```
Comment (tomxelliott) [inline charts/sptribs-wa-post-deployment-ft-tests/Chart.yaml]: line 12 also looks outdated
Should look like this:
```
  - name: java
    version: 5.2.1
    repository: 'https://hmctspublic.azurecr.io/helm/v1/repo/'
```

**Review by pr00279 — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/wapostdeploymentfttests/ScenarioRunnerTest.java [hunk 5]
```diff
@@ -118,7 +119,8 @@ public void setUp() {
     }
 
     @Test
-    public void scenarios_should_behave_as_specified() throws Exception {
+    @Ignore
```
Comment (oleon23) [inline src/functionalTest/java/uk/gov/hmcts/reform/wapostdeploymentfttests/ScenarioRunnerTest.java:122]: Does this mean all FT are not run if ScenarioRunnerTest is ignored?

File: build.gradle [hunk 6]
```diff
@@ -81,24 +81,15 @@ test {
 
 task functional(type: Test) {
   environment 'AZURE_SERVICE_BUS_CONNECTION_STRING','Endpoint=sb://ccd-servicebus-demo.servicebus.windows.net/;SharedAccessKeyName=SendAndReceiveCCDMessage;SharedAccessKey=BXOmN0uU5RZN7KW51soNUv7xNBcNtXqwh+ASbD8y4p4=;EntityPath=st-ccd-case-event'
```
Comment (oleon23) [inline build.gradle:83]: I think this needs to be updated to point to our equivalent access key. Though we don't have one in demo yet.
https://portal.azure.com/#@HMCTS.NET/resource/subscriptions/1c4f0704-a29e-403d-b719-b90c34ef14c9/resourceGroups/sptribs-aso-preview-rg/providers/Microsoft.ServiceBus/namespaces/sptribs-sb-preview/saskey

File: src/functionalTest/resources/application-functional.yaml (hunk 3 above)
Comment (oleon23) [inline src/functionalTest/resources/application-functional.yaml:18]: I think this needs to be updated once the `AZURE_SERVICE_BUS_CONNECTION_STRING` is updated.
Currently calling https://portal.azure.com/#@HMCTS.NET/resource/subscriptions/1c4f0704-a29e-403d-b719-b90c34ef14c9/resourceGroups/ccd-shared-aat/providers/Microsoft.ServiceBus/namespaces/ccd-servicebus-aat/topics/ccd-case-events-aat/overview

File: src/functionalTest/java/uk/gov/hmcts/reform/wapostdeploymentfttests/ScenarioRunnerTest.java (hunk 5 above)
Comment (gmmagruder) [reply src/functionalTest/java/uk/gov/hmcts/reform/wapostdeploymentfttests/ScenarioRunnerTest.java:122]: Yes the feature tests will not run with the ignore annotation. Currently the feature tests will fail but fixing them is out of scope for this ticket. This should be addressed in [DTSSTCI-577](https://tools.hmcts.net/jira/browse/DTSSTCI-577).

File: src/functionalTest/resources/application-functional.yaml (hunk 3 above)
Comment (oleon23) [reply src/functionalTest/resources/application-functional.yaml:18]: Decided to investigate further in [DTSSTCI-577](https://tools.hmcts.net/jira/browse/DTSSTCI-577).

File: build.gradle (hunk 6 above)
Comment (oleon23) [reply build.gradle:83]: Decided to investigate further in [DTSSTCI-577](https://tools.hmcts.net/jira/browse/DTSSTCI-577).

**Review by oleon23 — APPROVED** [review]
LGTM, further investigations around Servicebus can be reviewed as part of [DTSSTCI-577](https://tools.hmcts.net/jira/browse/DTSSTCI-577).

**Review by tomxelliott — APPROVED** [review]

**Pull request merged**